openforcefield/topology/molecule.py

```diff
@@ -3176,6 +3248,164 @@ def from_openeye(oemol, allow_undefined_stereo=False):
         return toolkit.from_openeye(
             oemol, allow_undefined_stereo=allow_undefined_stereo)
 
+    def to_qcshema(self):
+        """
+
+        :return:
+        """
+        raise NotImplementedError
+
+    @classmethod
+    def from_qcarchive(cls, qca_mol, smiles=None, allow_undefined_stereo=False):
+        """
+        Create a Molecule from  a QCArchive entry based on the cmiles implementation
+
+        Here there are two possible cases when the molecule has the bond order info in the connectivity data
+        and when it doesn't
+
+        Info always supplied, SMILES/InChi, elements, general connectivity and geometry.
+
+        Solution make a molecule from SMILES, then add the geometry as a conformer?
+        As the ordering is different we can use networkx to map the indices and rearange
+        this will also validate the info by checking that the general connection maps are the same
+
+        TODO do we get the same smiles string back?
+
+        Parameters
+        ----------
+        qca_mol : a QCArchive json representation
```

Comment:
If this is specifically the JSON representation string, we should name it something clearer, like `json` or `json_molecule`.
If we want to support alternative serializations (presuming that QCArchive may in the future), it may make more sense to accept a `dict` instead, since all serializations supported would presumably be reducible to a `dict`.

Reply:
For now, this is the `dict` representation with JSON encoding but we could also accept the actual record object as well and type check them to try and get the information out of it. Not sure what users would expect at this point though.

Reply:
Accepting both would be ideal, imho.